src/opm/output/eclipse/LoadRestart.cpp

```diff
@@ -1403,6 +1404,62 @@ namespace {
         smry.update(key("GITH"), xgrp[VI::XGroup::index::HistGasInjTotal]);
     }
 
+
+    void restore_udq(::Opm::SummaryState&             smry,
+                     const ::Opm::Schedule&           schedule,
+                     const std::shared_ptr<RestartFileView>& rst_view)
```

Comment:
This `const` is highly misleading.  The code only works because `rst_view` is (a reference to) a constant pointer to a mutable `RestartFileView`.  Calling `getKeyword<T>` will generally load data from disk which mutates the `RestartFileView` state.

Reply:
I'll remove the `const`- but I thought that `const std::shared_ptr<T>` just meant that the shared pointer itself would not be reseated - which is in line with the semantics implemented?

Reply:
> I'll remove the `const`- but I thought that `const std::shared_ptr<T>` just meant that the shared pointer itself would not be reseated - which is in line with the semantics implemented?

That's fine if that's the intended semantics.  Please note, however, that other functions take a `shared_ptr<RestartFileView>` by value which also has the same semantics from the callers point of view and does lead the casual reader to assume that the `RestartFileView` contained in the `shared_ptr<>` itself is `const`.  Finally, taking `shared_ptr<>` (or `unique_ptr<>` for that matter) by reference is usually bad practice.